app/models/admin_dashboard_data.rb

```diff
@@ -3,17 +3,46 @@
 class AdminDashboardData
   include StatsCacheable
 
+  class Problem
+    VALID_PRIORITIES = ["low", "high"].freeze
+
+    attr_reader :message, :priority, :identifier
+
+    def initialize(message, priority: "low", identifier: nil)
+      @message = message
+      @priority = VALID_PRIORITIES.include?(priority) ? priority : "low"
+      @identifier = identifier
+    end
+
+    def to_s
+      @message
+    end
+
+    def to_h
+      { message: message, priority: priority, identifier: identifier }
+    end
+
+    def self.from_h(h)
+      h = h.with_indifferent_access
+      Problem.maybe_create(h[:message], priority: h[:priority], identifier: h[:identifier])
```

Comment:
What do you think about having `initialize` check that `message` is not a blank string instead? I feel like that might be better because I don't think `Problem` works well with a blank string at all. For example, you wouldn't expect `Problem#to_s` to return `nil`.